akka-javasdk/src/main/java/akka/javasdk/http/QueryParams.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.http;
+
+
+import akka.http.javadsl.model.Query;
+import akka.japi.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Represents the query parameters of a request.
+ */
+public record QueryParams(Query query) {
+
+  /**
+   * Returns the value of the first parameter with the given key if it exists.
+   */
+  public Optional<String> get(String key){
+    return query.get(key);
+  }
+
+  /**
+   * Returns the Integer value of the first parameter with the given key if it exists.
+   */
+  public Optional<Integer> getInteger(String key){
+    return query.get(key).map(Integer::parseInt);
+  }
+
+  /**
+   * Returns the Long value of the first parameter with the given key if it exists.
+   */
+  public Optional<Long> getLong(String key) {
+    return query.get(key).map(Long::parseLong);
+  }
+
+  /**
+   * Returns the Boolean value of the first parameter with the given key if it exists.
+   */
+  public Optional<Boolean> getBoolean(String key) {
+    return query.get(key).map(Boolean::parseBoolean);
+  }
+
+  /**
+   * Returns the Double value of the first parameter with the given key if it exists.
+   */
+  public Optional<Double> getDouble(String key) {
+    return query.get(key).map(Double::parseDouble);
+  }
+
+  /**
+   * Returns the Float value of the first parameter with the given key if it exists.
+   */
+  public Optional<Float> getFloat(String key) {
+    return query.get(key).map(Float::parseFloat);
+  }
+
+  /**
+   * Returns the Short value of the first parameter with the given key if it exists.
+   */
+  public Optional<Short> getShort(String key) {
```

Comment:
We could skip the exotic types, in my opinion. Float can be covered by Double. Short by Integer. Character by String.